src/lib/tls/tls13/tls_server_impl_13.cpp

```diff
@@ -45,13 +45,25 @@ std::string Server_Impl_13::application_protocol() const {
 std::vector<X509_Certificate> Server_Impl_13::peer_cert_chain() const {
    if(m_resumed_session.has_value()) {
       return m_resumed_session->peer_certs();
-   } else if(m_handshake_state.has_client_certificate_chain()) {
+   } else if(m_handshake_state.has_client_certificate_msg() &&
+             m_handshake_state.client_certificate().has_certificate_chain()) {
       return m_handshake_state.client_certificate().cert_chain();
    } else {
       return {};
    }
 }
 
+std::shared_ptr<const Public_Key> Server_Impl_13::peer_raw_public_key() const {
+   if(m_resumed_session.has_value()) {
+      return m_resumed_session->peer_raw_public_key();
```

Comment:
Is there a reason the server prefers the resumed value other the current one, and the client does it the other way around?

Reply:
The order doesn't really matter: A resumed session does not have a "current" certificate. When resuming a session, the server won't authenticate again. And vice-versa: when we have a Certificate message, we are not in a resumed session.

I.e. the order doesn't really matter. Though I'll fix it for consistency, thanks.